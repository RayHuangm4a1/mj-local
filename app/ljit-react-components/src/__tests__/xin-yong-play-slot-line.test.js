import React from 'react';
import { shallow, mount, } from 'enzyme';
import XinYongPlaySlotLine from '../components/xin-yong-play-slot/line';

const { TypeEnums, } = XinYongPlaySlotLine;

jest.mock('antd/lib/tooltip');

describe('XinYongPlaySlotLine', () => {
	let play;
	let betting;

	beforeEach(() => {
		play = { name: '大', id: 53000, odds: 1954.00, };
		betting = {
			play: { name: '大', id: 53000, odds: 1954.00, },
			amount: 10,
		};
	});

	afterEach(() => {
		play = {};
		betting = {};
	});

	it('handle default props', () => {
		const {
			isDisabled,
			disabledText,
			defaultAmount,
			onChange,
			play,
			playSlotType,
			betting,
		} = XinYongPlaySlotLine.defaultProps;

		expect(isDisabled).toEqual(false);
		expect(disabledText).toEqual('不支援');
		expect(defaultAmount).toEqual(0);
		expect(play).toMatchObject({});
		expect(playSlotType).toEqual(TypeEnums.CIRCLE);
		expect(betting).toMatchObject({});
		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const props = {
			play: play,
			betting: betting,
			className: 'mock-class',
			isDisabled: false,
			disabledText: 'mock-text',
			defaultAmount: 0,
			onChange: () => {},
		};

		const wrapper = shallow(<XinYongPlaySlotLine {...props}/>);

		expect(wrapper).toMatchSnapshot();
	});

	it ('should be selectable by class ljit-xin-yong-play-slot-line', () => {
		const props = {
			play: play,
			betting: betting,
		};
		const wrapper = shallow(<XinYongPlaySlotLine {...props} />);

		expect(wrapper.hasClass('ljit-xin-yong-play-slot-line')).toEqual(true);
	});

	it ('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<XinYongPlaySlotLine className={className} play={play} betting={betting} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it ('should be selectable by class ljit-xin-yong-play-slot-line--disabled, if isDisabled is pass true', () => {
		const props = {
			play: play,
			betting: betting,
			isDisabled: true,
		};
		const wrapper = shallow(<XinYongPlaySlotLine {...props} />);

		expect(wrapper.hasClass('ljit-xin-yong-play-slot-line--disabled')).toEqual(true);
	});

	it('should handle onChange event', () => {
		const onChange = jest.fn();
		const mockedEvent = { target: {}, };

		const wrapper = shallow(
			<XinYongPlaySlotLine defaultAmount={10} onChange={onChange} play={play} betting={betting} />
		);

		wrapper.simulate('click', mockedEvent);

		expect(onChange).toBeDefined();
		expect(onChange).toHaveBeenCalledTimes(1);
	});

	it ('should mount in a full DOM', () => {
		const isDisabled = false;
		const disabledText = 'mock-text';
		const defaultAmount = 0;
		const className = 'mock-class';
		const onChange = () => {};
		const wrapper = mount(
			<XinYongPlaySlotLine
				play={play}
				betting={betting}
				isDisabled={isDisabled}
				disabledText={disabledText}
				defaultAmount={defaultAmount}
				onChange={onChange}
				className={className}
			/>
		);

		expect(wrapper.props().play).toMatchObject(play);
		expect(wrapper.props().betting).toMatchObject(betting);
		expect(wrapper.props().isDisabled).toEqual(isDisabled);
		expect(wrapper.props().disabledText).toEqual(disabledText);
		expect(wrapper.props().defaultAmount).toEqual(defaultAmount);
		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().onChange).toEqual(onChange);
	});
});

describe('XinYongPlaySlotLine TypeEnums', () => {
	it('should contains CIRCLE property', () => {
		const typeName = 'circle';
		const formatType = 'CIRCLE';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});

	it('should contains ROUND property', () => {
		const typeName = 'round';
		const formatType = 'ROUND';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});

	it('should contains RECTANGLE property', () => {
		const typeName = 'rectangle';
		const formatType = 'RECTANGLE';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});

	it('should contains ANIMAL property', () => {
		const typeName = 'animal';
		const formatType = 'ANIMAL';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});
});
