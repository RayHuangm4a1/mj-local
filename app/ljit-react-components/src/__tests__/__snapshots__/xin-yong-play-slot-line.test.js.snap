// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XinYongPlaySlotLine should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <XinYongPlaySlotLine
    betting={
      Object {
        "amount": 10,
        "play": Object {
          "id": 53000,
          "name": "大",
          "odds": 1954,
        },
      }
    }
    className="mock-class"
    defaultAmount={0}
    disabledText="mock-text"
    isDisabled={false}
    onChange={[Function]}
    play={
      Object {
        "id": 53000,
        "name": "大",
        "odds": 1954,
      }
    }
    playSlotType="circle"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ljit-xin-yong-play-slot-line__ball"
        >
          <Circle
            fontSize="extra-small"
            size="extra-small"
            text="大"
            type="selected"
          />
        </div>,
        <div
          className="ljit-xin-yong-play-slot-line__odds"
        >
          <ControlledTooltip
            isVisible={false}
            onVisibleChange={[Function]}
            overlayClassName="ljit-xin-yong-play-slot-line__overlay"
            overlayColor="white"
            placement="top"
            title={
              <DecimalNumber
                className=""
                data={1954}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
            }
          >
            <div>
              <DecimalNumber
                className=""
                data={1954}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
            </div>
          </ControlledTooltip>
        </div>,
        <div
          className="ljit-xin-yong-play-slot-line__status"
        >
          <Input
            className="ljit-xin-yong-play-slot-line__input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onPressEnter={[Function]}
            readOnly={false}
            value="10"
          />
        </div>,
      ],
      "className": "ljit-xin-yong-play-slot-line ljit-xin-yong-play-slot-line--checked mock-class",
      "onClick": [Function],
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Circle
            fontSize="extra-small"
            size="extra-small"
            text="大"
            type="selected"
          />,
          "className": "ljit-xin-yong-play-slot-line__ball",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fontSize": "extra-small",
            "size": "extra-small",
            "text": "大",
            "type": "selected",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ControlledTooltip
            isVisible={false}
            onVisibleChange={[Function]}
            overlayClassName="ljit-xin-yong-play-slot-line__overlay"
            overlayColor="white"
            placement="top"
            title={
              <DecimalNumber
                className=""
                data={1954}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
            }
          >
            <div>
              <DecimalNumber
                className=""
                data={1954}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
            </div>
          </ControlledTooltip>,
          "className": "ljit-xin-yong-play-slot-line__odds",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              <DecimalNumber
                className=""
                data={1954}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
            </div>,
            "isVisible": false,
            "onVisibleChange": [Function],
            "overlayClassName": "ljit-xin-yong-play-slot-line__overlay",
            "overlayColor": "white",
            "placement": "top",
            "title": <DecimalNumber
              className=""
              data={1954}
              decimalPlaces={4}
              hasSeparator={true}
              isCurrency={false}
              isPercent={false}
              isTolerance={false}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <DecimalNumber
                className=""
                data={1954}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />,
            },
            "ref": [Function],
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "data": 1954,
                "decimalPlaces": 4,
                "hasSeparator": true,
                "isCurrency": false,
                "isPercent": false,
                "isTolerance": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Input
            className="ljit-xin-yong-play-slot-line__input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onPressEnter={[Function]}
            readOnly={false}
            value="10"
          />,
          "className": "ljit-xin-yong-play-slot-line__status",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "ljit-xin-yong-play-slot-line__input",
            "disabled": false,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "onPressEnter": [Function],
            "readOnly": false,
            "value": "10",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ljit-xin-yong-play-slot-line__ball"
          >
            <Circle
              fontSize="extra-small"
              size="extra-small"
              text="大"
              type="selected"
            />
          </div>,
          <div
            className="ljit-xin-yong-play-slot-line__odds"
          >
            <ControlledTooltip
              isVisible={false}
              onVisibleChange={[Function]}
              overlayClassName="ljit-xin-yong-play-slot-line__overlay"
              overlayColor="white"
              placement="top"
              title={
                <DecimalNumber
                  className=""
                  data={1954}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />
              }
            >
              <div>
                <DecimalNumber
                  className=""
                  data={1954}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />
              </div>
            </ControlledTooltip>
          </div>,
          <div
            className="ljit-xin-yong-play-slot-line__status"
          >
            <Input
              className="ljit-xin-yong-play-slot-line__input"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onPressEnter={[Function]}
              readOnly={false}
              value="10"
            />
          </div>,
        ],
        "className": "ljit-xin-yong-play-slot-line ljit-xin-yong-play-slot-line--checked mock-class",
        "onClick": [Function],
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Circle
              fontSize="extra-small"
              size="extra-small"
              text="大"
              type="selected"
            />,
            "className": "ljit-xin-yong-play-slot-line__ball",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "fontSize": "extra-small",
              "size": "extra-small",
              "text": "大",
              "type": "selected",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ControlledTooltip
              isVisible={false}
              onVisibleChange={[Function]}
              overlayClassName="ljit-xin-yong-play-slot-line__overlay"
              overlayColor="white"
              placement="top"
              title={
                <DecimalNumber
                  className=""
                  data={1954}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />
              }
            >
              <div>
                <DecimalNumber
                  className=""
                  data={1954}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />
              </div>
            </ControlledTooltip>,
            "className": "ljit-xin-yong-play-slot-line__odds",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div>
                <DecimalNumber
                  className=""
                  data={1954}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />
              </div>,
              "isVisible": false,
              "onVisibleChange": [Function],
              "overlayClassName": "ljit-xin-yong-play-slot-line__overlay",
              "overlayColor": "white",
              "placement": "top",
              "title": <DecimalNumber
                className=""
                data={1954}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <DecimalNumber
                  className=""
                  data={1954}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />,
              },
              "ref": [Function],
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "data": 1954,
                  "decimalPlaces": 4,
                  "hasSeparator": true,
                  "isCurrency": false,
                  "isPercent": false,
                  "isTolerance": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Input
              className="ljit-xin-yong-play-slot-line__input"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onPressEnter={[Function]}
              readOnly={false}
              value="10"
            />,
            "className": "ljit-xin-yong-play-slot-line__status",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "ljit-xin-yong-play-slot-line__input",
              "disabled": false,
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "onPressEnter": [Function],
              "readOnly": false,
              "value": "10",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
